### ISSUE TITLE
AutoUpload view crashes app

### ISSUE THREAD

**Issue by despens** [nextcloud/android#1100]
### Actual behaviour
- Switching to instant view upload crashes the app. 
- No files are ever auto-uploaded

### Expected behaviour
- Switching to instant upload view should show instant upload view. 
- Files should be auto-uploaded as configured
 
### Steps to reproduce
1. Start nextcloud app
2. Select Instant Upload view
3. 💥 

### Environment data
Android version: 6.0.1

Device model: STV100-1 (BlackBerry Priv)

Stock or customized system: stock

Nextcloud app version: 1.4.3

Nextcloud server version: 11.0.3

PR: #1201

**Comment by tobiasKaminsky** [comment]
Is this really "instantupload" and not "AutoUpload" as you are using Android 6?
Can you provide us with a logcat?

**Comment by despens** [comment]
Yes it is indeed "AutoUpload", just the directory on my nextcloud is called "InstantUpload". I don't know what a logcat is, if you let me know I will try to provide.

**Comment by tobiasKaminsky** [comment]
Unfortunately there is no easy way to get a proper error message/logcat: https://wiki.lineageos.org/logcat.html

**Comment by despens** [comment]
Find my logcat here: https://nc.1x-upon.com/s/GA5u0VDJ155hLvT
I hope it is useful.

**Comment by tobiasKaminsky** [comment]
Thanks for the logcat, but I cannot see anything related to nextcloud. Please try it again and open autoupload view / nextcloud 2-3 times and verify that in the logcat is something with "nextcloud" and "crash" or similar.

**Comment by creideiki** [comment]
I'm seeing the same thing on my phone:

- Fairphone 2
- Fairphone Open 17.06.4 (based on Android 6.0.1)
- Nextcloud app 1.4.2 (from F-Droid)
- Nextcloud server 12.0.0

The newest file in my `InstantUpload` folder is from 2017-06-10. Shortly after, there were new releases of both the Nextcloud app (1.4.2 on 2017-06-13) and the OS (Fairphone Open 17.06.4 on 2017-06-15, which was an upgrade from Android 5.1 to 6.0).

Running `adb logcat` while clicking "Auto upload" in the app gives this part that seems relevant:
```
07-16 17:30:12.391  2270  2445 I ActivityManager: Displayed com.nextcloud.client/com.owncloud.android.ui.activity.FolderSyncActivity: +248ms
--------- beginning of crash
07-16 17:30:12.422  4370 32348 E AndroidRuntime: FATAL EXCEPTION: Thread-77
07-16 17:30:12.422  4370 32348 E AndroidRuntime: Process: com.nextcloud.client, PID: 4370
07-16 17:30:12.422  4370 32348 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.lastIndexOf(java.lang.String)' on a null object reference
07-16 17:30:12.422  4370 32348 E AndroidRuntime:        at com.owncloud.android.datamodel.MediaProvider.getMediaFolders(MediaProvider.java:89)
07-16 17:30:12.422  4370 32348 E AndroidRuntime:        at com.owncloud.android.ui.activity.FolderSyncActivity$1.run(FolderSyncActivity.java:154)
07-16 17:30:12.422  4370 32348 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:818)
07-16 17:30:12.435  2270  3491 W ActivityManager:   Force finishing activity com.nextcloud.client/com.owncloud.android.ui.activity.FolderSyncActivity
07-16 17:30:12.439  4370  4370 D FolderSyncActivity: onPause() ending
07-16 17:30:12.439  2270  3491 W ActivityManager:   Force finishing activity com.nextcloud.client/com.owncloud.android.ui.activity.FileDisplayActivity
07-16 17:30:12.476  4370  4370 D FileDisplayActivity: onStop() ending
07-16 17:30:12.478  4370  4370 D FileDisplayActivity: onDestroy() ending
07-16 17:30:12.486  2270  3309 I OpenGLRenderer: Initialized EGL, version 1.4
07-16 17:30:12.488  4370  4370 V FileDownloader: Destroying service
07-16 17:30:12.490  4370  4370 V FileUploader: Destroying service
```

Trying to downgrade the app to 1.4.1 unfortunately doesn't work; F-Droid says "Failed to install due to an unknown error", and no further information is available with `adb logcat`.

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger I am unsure why the filePath is null. Is it safe to "just" omit and remove the null-items? You are getting them via contentResolver, so they should not be null...?

**Comment by creideiki** [comment]
When I take photos and immediately see that they came out wrong, I usually delete them right away. Sometimes, I manage to delete photos before auto upload has a chance to get to them, which at least in previous versions of the app (when I could see the auto upload tab without it crashing) made it confused. Might that have something to do with this problem?

**Comment by AndyScherzinger** [comment]
@creideiki can you check if this also happens with the dev version?
@tobiasKaminsky that shouldn't happen, right. Only thing I could think of is if the media index is obsolete as in the files being indexed but deleted while it hasn't been reported back to the index and only the next re-indexing will correct this. Omitting them should thus be fine while it is not that simple since we still want to show 8 files and the total number of files indication.

**Comment by creideiki** [comment]
It does, and the crash log looks similar. This is with Nextcloud dev 20170713 from F-Droid:
```
07-17 13:11:25.697   581  1745 E AndroidRuntime: FATAL EXCEPTION: Thread-551
07-17 13:11:25.697   581  1745 E AndroidRuntime: Process: com.nextcloud.android.beta, PID: 581
07-17 13:11:25.697   581  1745 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.lastIndexOf(java.lang.String)' on a null object reference
07-17 13:11:25.697   581  1745 E AndroidRuntime:        at com.owncloud.android.datamodel.MediaProvider.getMediaFolders(MediaProvider.java:126)
07-17 13:11:25.697   581  1745 E AndroidRuntime:        at com.owncloud.android.ui.activity.FolderSyncActivity$1.run(FolderSyncActivity.java:180)
07-17 13:11:25.697   581  1745 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:818)
07-17 13:11:25.699  1996  2013 W ActivityManager:   Force finishing activity com.nextcloud.android.beta/com.owncloud.android.ui.activity.FolderSyncActivity
07-17 13:11:25.712  1996  2013 W ActivityManager:   Force finishing activity com.nextcloud.android.beta/com.owncloud.android.ui.activity.FileDisplayActivity
```

**Comment by tobiasKaminsky** [comment]
> Omitting them should thus be fine while it is not that simple since we still want to show 8 files and the total number of files indication.

If a filepath is null it will be skipped, both for total numbers and for counting the showed numbers.
--> #1201

**Comment by AndyScherzinger** [comment]
merged to master for 1.5.0
